sw/device/silicon_creator/lib/drivers/alert.c

```diff
@@ -0,0 +1,175 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/alert.h"
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#include "alert_handler_regs.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+enum {
+  kBase = TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR,
+};
+
+rom_error_t alert_configure(size_t index, alert_class_t cls,
+                            alert_enable_t enabled) {
+  if (index >= ALERT_HANDLER_ALERT_CLASS_MULTIREG_COUNT) {
+    return kErrorAlertBadIndex;
+  }
+  index *= 4;
+
+  switch (cls) {
+    case kAlertClassA:
+      abs_mmio_write32(kBase + ALERT_HANDLER_ALERT_CLASS_0_REG_OFFSET + index,
+                       0);
+      abs_mmio_write32(
+          kBase + ALERT_HANDLER_LOC_ALERT_CLASS_0_REG_OFFSET + index, 0);
```

Comment:
This doesn't seem right, I would expect the local alerts to be handled separately. According to the HW spec, alert handler has only 4 local alerts: alert pingfail, excalation pingfail, alert integfail, and escalation integfail. We may want to reserve the first or last 4 alerts to local alerts if we want to use the same range for both types of alerts.

Reply:
After reading the spec some more and speaking to msf@ about local alerts, I'm going to separate configuring local alerts into its own version of this function.  The `ALERT_CLASS_LOC_*` register offsets do not universally follow (in register number order) the non-local alerts and the alert manager is going to gain more local alerts.

I think its best to separate the configuration functions so we wouldn't have to rely on register ordering _and_ so we don't have to create an alert numbering scheme that accommodates local and non-local alerts with a single integer index.

Reply:
I agree, thanks.